feat(hooks): Add useOnKeyDown hook (#396)
GetJobber/atlantis: packages/hooks/src/useOnKeyDown/useOnKeyDown.mdx

---
name: useOnKeyDown
menu: Hooks
route: /hooks/use-on-key-down
---

import { Playground } from "docz";
import { useState } from "react";
import { useOnKeyDown } from "./useOnKeyDown";
import { Text } from "@jobber/components/Text";

# UseOnKeyDown

> `useOnKeyDown` is an extremly simple hook that can be for adding the keydown event handler when the component is mounted and remove them when it unmount.

Reviewer: Little wording thought. 😄 

```suggestion
`useOnKeyDown` is a simple hook adding the keydown event handler when the component is mounted and removing when unmounted.
```

> `useOnKeyDown` should **only** only by used when making keyboard shortcut in any component.

Reviewer: Another little wording thought. 😄 

```suggestion
`useOnKeyDown` should **only ** be used when making keyboard shortcut in a component.
```

```tsx
import { useOnKeyDown } from "@jobber/hooks";
```

<Playground>
  {()=>{

>     const letterText = "Delete me by pressing Delete key on your keyboard.";

Reviewer: Macs label backspace and delete quite oddly which make this example a bit confusing for users on that platform. We can probably get a way with simplifying this just down to a single example. How would you feel about removing this delete one and switching your second one to `escape` `space-enter` and something like `ctrl-i` or something?

    const letterText2 = "Space Enter and Shift + Escape will show here";
    const [ text, setText ] = useState(letterText);
    const [ textTwo, setTextTwo ] = useState(letterText2);
    useOnKeyDown(() => setText("Deleted. Press Control + z for undo."), "Delete");
    useOnKeyDown(() => setText(letterText), {key: "z", ctrlKey: true});
    useOnKeyDown((e) => {
        if(e.shiftKey) setTextTwo("You pressed Shift and "+e.key);
        else setTextTwo("You pressed '"+e.key+"'");
    }, [" ", "Enter", {key: "Escape", shiftKey: true} ]);

>     return (<Text>{text}<pre />{textTwo}</Text>);}}

Reviewer: Looks like we have an extra `<pre />` here.

Reviewer: Might also be nice to keep the instructions fixed and have a second line that displays the result of the key events.
